mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOpsTypes.h

```diff
@@ -86,6 +86,9 @@ template <typename Parameter> class SYCLInheritanceTypeTrait {
 /// Return true if the given \p Ty is a SYCL type.
 inline bool isSYCLType(Type Ty) { return isa<SYCLDialect>(Ty.getDialect()); }
 
+/// Return the number of dimensions of type \p Ty.
+unsigned getDimensions(Type Ty);
```

Comment:
What does this function return if the given type is not a type that has a dimension? I am leaning toward asserting in those cases and returning 0.

Reply:
It has an implicit `assert`, so it will fail